cfgen/cfgen

```diff
@@ -476,9 +476,23 @@ def run_command(hostname: str, *args: str) -> str:
                                    timeout=15).decode()
 
 
+def validate_facter(hostname: str) -> None:
```

Comment:
1. It's a double check. Moreover, Hare may have been installed from sources, so RPM-level requirements don't always apply.
2. I am not sure what other dependencies need to be verified. For instance,
   1. Dhall is installed explicitly by Hare. We use very specific Dhall version but as long as our Dhall files are processed (i.e. the compiler doesn't complain on syntax), we're good.
   2. Python-level dependencies are handled by pip package manager. We already have PR #1688 that locks pythonic dependencies

Reply:
Thanks for explaining
Consul validation?

Reply:
Probably. But that check doesn't need to be performed from cfgen. 

@mssawant What's your opinion?

Reply:
May be we can add these checks (including facter) as part of build (Makefile) itself instead of cfgen. Let the build itself fail before invoking cfgen.

Reply:
make is run at the build machine to create an RPM. That RPM is shipped afterwards to the field. So I don't think it makes sense to check the versions from the Makefile, those checks have nothing to do with the runtime environment anyways.

Reply:
@suvratjoshi I think I've recalled one use case when this PR is needed and why RPM-level restriction for facter is probably not sufficient.

The RPM checks are always local. So they apply to the RPM packages installed at this particular machine (i.e. where `rpm -i` or `yum install` is invoked). But cfgen goes to another machines where some older RPM packages can be installed (e.g. because of human mistake). There is no guarantee that the remote machine has the proper facter version installed.